### PR TITLE
add siac host commands

### PR DESCRIPTION
**Pull request by lukechampine**

### CODE REVIEW

File: siac/hostcmd.go [hunk 1]
```diff
@@ -45,8 +49,137 @@ You may also supply a specific address to be announced, e.g.:
...
+		return
+	}
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, 4, ' ', 0)
+	fmt.Fprintln(w, "\nUsed\tCapacity\t% Used\tPath")
+	for _, folder := range sg.StorageFolderMetadata {
+		curSize := int64(folder.Capacity - folder.CapacityRemaining)
+		pctUsed := 100 * (float64(curSize) / float64(folder.Capacity))
+		fmt.Fprintf(w, "%s\t%s\t%.2f\t%s\n", filesizeUnits(curSize), filesizeUnits(int64(folder.Capacity)), pctUsed, folder.Path)
+	}
+	w.Flush()
```
Comment (lukechampine) [inline siac/hostcmd.go]: Example output:
```
Host info:
	Storage:      20.000 GB (0 B used)
	Price:        19999 SC per TB per month
	Max Duration: 25920 Blocks

	Accepting Contracts: No
	Anticipated Revenue: 0
	Revenue:             0
	Lost Revenue:        0

	Net Address: 

RPC Stats:
	Error Calls:        0
	Unrecognized Calls: 1
	Download Calls:     0
	Renew Calls:        0
	Revise Calls:       0
	Settings Calls:     0
	FormContract Calls: 0

Used    Capacity     % Used    Path
0 B     10.000 GB    0.00      /home/luke/foo
0 B     10.000 GB    0.00      /home/luke/bar
```

File: siac/hostcmd.go [hunk 2]
```diff
@@ -45,8 +49,137 @@ You may also supply a specific address to be announced, e.g.:
...
+		Long: `Change how much data a storage folder should store. If the new size is less
+than what the folder is currently storing, data will be distributed across the
+other storage folders.`,
+		Run: wrap(hostfolderresizecmd),
+	}
+
+	hostSectorCmd = &cobra.Command{
+		Use:   "sector",
+		Short: "Add or delete a sector",
+		Long:  "Add or delete a sector. Note that deleting a sector may impact host revenue.",
```
Comment (DavidVorick) [inline siac/hostcmd.go]: There is no 'add sector' support right now.

Comment (lukechampine) [inline siac/hostcmd.go]: true, but I felt like just `Delete a sector` would be misleading.

File: siac/hostcmd.go (hunk 1 above)
Comment (DavidVorick) [inline siac/hostcmd.go]: Example output:
```
Host info:
	Storage:      20.000 GB (0 B used)
	Price:        19999 SC per TB per month
	Max Duration: 25920 Blocks

	Accepting Contracts: No
	Anticipated Revenue: 0
	Revenue:             0
	Lost Revenue:        0

	Net Address: 

RPC Stats:
	Error Calls:        0
	Unrecognized Calls: 1
	Download Calls:     0
	Renew Calls:        0
	Revise Calls:       0
	Settings Calls:     0
	FormContract Calls: 0

Storage Stats:
    Used    Capacity     % Used    Path
    0 B     10.000 GB    0.00      /home/luke/foo
    0 B     10.000 GB    0.00      /home/luke/bar
```

Comment (DavidVorick) [inline siac/hostcmd.go]: Jordan does not like my example

File: siac/hostcmd.go [hunk 3]
```diff
@@ -45,8 +49,137 @@ You may also supply a specific address to be announced, e.g.:
...
+		Long: `Delete a sector, identified by its Merkle root. Note that deleting a
+sector may impact host revenue.`,
+		Run: wrap(hostsectordeletecmd),
+	}
 )
 
+// hostcmd is the handler for the command `siac host`.
+// Prints info about the host and its storage folders.
+func hostcmd() {
+	hg := new(api.HostGET)
```
Comment (VoidingWarranties) [inline siac/hostcmd.go]: Why use new here instead of `var hg api.HostGET`?

File: siac/hostcmd.go [hunk 4]
```diff
@@ -45,8 +49,137 @@ You may also supply a specific address to be announced, e.g.:
...
+	accept := yesNo(is.AcceptingContracts)
+	// convert price to SC/TB/mo
+	price, err := modules.StoragePriceToHuman(is.MinimumStoragePrice)
+	if err != nil {
+		price = ^uint64(0)
+	}
+	totalRevenue := fm.ContractCompensation.
+		Add(fm.StorageRevenue).
+		Add(fm.DownloadBandwidthRevenue).
+		Add(fm.UploadBandwidthRevenue)
```
Comment (VoidingWarranties) [inline siac/hostcmd.go]: I didn't know you could do this, cool!

File: siac/hostcmd.go [hunk 5]
```diff
@@ -45,8 +49,137 @@ You may also supply a specific address to be announced, e.g.:
...
+		totalstorage += folder.Capacity
+		storageremaining += folder.CapacityRemaining
+	}
+
+	// convert accepting bool
+	accept := yesNo(is.AcceptingContracts)
+	// convert price to SC/TB/mo
+	price, err := modules.StoragePriceToHuman(is.MinimumStoragePrice)
+	if err != nil {
+		price = ^uint64(0)
```
Comment (VoidingWarranties) [inline siac/hostcmd.go]: Why do we use the max price on error?

File: siac/hostcmd.go (hunk 3 above)
Comment (DavidVorick) [inline siac/hostcmd.go]: `new` gives you a pointer, if you do `var hg api.HostGET` you get the actual object, and if you do `var hg *api.HostGET` you get a nil pointer. The way above gives you a pointer to an object that is initialized.

File: siac/hostcmd.go (hunk 5 above)
Comment (DavidVorick) [inline siac/hostcmd.go]: There should be a comment here, or better yet an explicit error check. Right now, the only error that can be returned by the `StoragePriceToHuman` function is if there are more coins than can be represented by a uint64, but that's not obvious just by looking at this code.

File: siac/hostcmd.go (hunk 4 above)
Comment (DavidVorick) [inline siac/hostcmd.go]: yeah it was new to me too

File: siac/hostcmd.go (hunk 3 above)
Comment (lukechampine) [inline siac/hostcmd.go]: you don't need a pointer though. In most cases we should be doing `var hg api.HostGET`.

File: siac/hostcmd.go (hunk 4 above)
Comment (lukechampine) [inline siac/hostcmd.go]: it's more common in dot-heavy languages like javascript. They like to abuse this pattern to chain tons and tons of actions together.

File: siac/hostcmd.go (hunk 3 above)
Comment (VoidingWarranties) [inline siac/hostcmd.go]: But here you are both using `new` and using `&hg` to take the address of the value when passing it to `getAPI`. Isn't this a pointer to a pointer? Why is this necessary?

File: siac/hostcmd.go (hunk 5 above)
Comment (VoidingWarranties) [inline siac/hostcmd.go]: +1 for explicit error check

File: siac/hostcmd.go (hunk 3 above)
Comment (lukechampine) [inline siac/hostcmd.go]: huh. Right you are. JSON's unmarshalling rules prevent this causing problems, but it's definitely wrong
